### ISSUE TITLE
Deovr and Heresphere both stuttering

### ISSUE THREAD

**Issue by onmlul** [xbapps/xbvr#1567]
Currently any video I have will have a stutter every few seconds, making them unwatchable.
I'm using the quest 3 and all videos run perfectly fine in both applications in Steamlink. It's specifically whern I run them through Quest while using XBVR. I'm using 5 ghz wifi with little to no interference if that matters.

time="2024-01-04T15:41:35-06:00" level=error msg="dial tcp 192.168.0.235:23554: connectex: No connection could be made because the target machine actively refused it."

Seems to be filling the log file with every lag.

**Comment by pops64** [comment]
I use to have this issue but found it to be an issue with the host and not XBVR. If system resource demand is high XBVR doesn't have priority to files. I found quitting Heresphere and reloading Heresphere fixed this issue or shuting down what ever was consuming system resources. This is confirmed by @4gnusd3i response as those types of files require almost no delay in sending the file out. I am now on a Media Server with ZFS housing my files and I run my Quest off the same band has my Air Link connection(Only Quest is on it) and I am 5 feet from the router and a 2.5 Gbps connection from the router to the server.  

You got to remember the bit rate of the file is at the bare minimum the required speed of the connection and "Original" files are at 80+ Mbps as XBVR direct streams files. When I had the issue you were describing I was still on my PC with an external drive. I could easily replicate it if the external drive was sharing resources. 

If you know your way around encoding. Trying lowering the bit rate or find a file with a lower bit rate

**Comment by VictoriousJuniper** [comment]
I have the same issue with files stuttering in Heresphere on Quest3, but it is very inconsistent. I've had it when using a USB 3.2 drive connected to a Mac Studio running the XBVR docker container, but now also on a new NAS using a 5 disk ZFS RAIDZ2 pool.

The thing is, when the stuttering happens, it happens a few seconds into the scene. When I then stop the playback and start the same scene again from the main menu, the stuttering is gone. Then on other files, it randomly comes back. My NAS has an i9-13900HX, 96GB RAM, 1 Gbit wired connection. My Quest 3 reports 1200mbit bandwidth on 802.11ac. Other players/servers on the same NAS have no issues with 90-100GB BR Remuxes. So, I'm not sure if it's a Herespere, XBVR or Quest 3 issue.

**Comment by pops64** [comment]
> @pops64 I got an email with the following reply from you, not sure why it doesnt show;
> 
> > _
> > I use to have this issue but found it to be an issue with the host and not XBVR. If system resource demand is high XBVR doesn't have priority to files. I found quitting Heresphere and reloading Heresphere fixed this issue or shuting down what ever was consuming system resources. This is confirmed by [@4gnusd3i](https://github.com/4gnusd3i) response as those types of files require almost no delay in sending the file out. I am now on a Media Server with ZFS housing my files and I run my Quest off the same band has my Air Link connection(Only Quest is on it) and I am 5 feet from the router and a 2.5 Gbps connection from the router to the server.
> > You got to remember the bit rate of the file is at the bare minimum the required speed of the connection and "Original" files are at 80+ Mbps as XBVR direct streams files. When I had the issue you were describing I was still on my PC with an external drive. I could easily replicate it if the external drive was sharing resources.
> > If you know your way around encoding. Trying lowering the bit rate or find a file with a lower bit rate
> > _
> 
> Even though I am in agreement that this is most of the time a host issue, I am not sure that it is in my case. I am running my files of a 4x8TB DAS in RAID10, and have no problems watching the files on my system connected over USB3. However if I stream it over XBVR it will stutter every second over a 5GHz connection of 500/500Mb/s. This might be an issue with DeoVR or the Quest 2 itself, however I thought it might be worth a mention, if this could be XBVR related. As I said, it only happens with files commonly above 60GB/Hr, often 100GB/Hr roughly.

Yeah I deleted my original post. I thought more about it and wasn't 100% sure on it. With that being said for me to remove the stutter. I set the here sphere buffer to 5 secs from 3(I can't recall which one. I think its the one for playback start buffer) which seems to get ride of the problem when I recreate due to load on the server(Downloading files while watching at the same time).

There are lots of things that could cause a bottleneck. While I am not 100% sure it is XBVR causing it, I couldn't say for sure with the sheer amounts of different configurations.
